src/core_plugins/console/public/tests/src/content_type.js

```diff
@@ -0,0 +1,59 @@
+import { getContentType } from '../../src/es';
+
+const { test, module, equal } = window.QUnit;
+
+const APPLICATION_NEWLINE_JSON = 'application/x-ndjson';
+const APPLICATION_JSON = 'application/json';
+
+module('Content type');
+
+test('line delimited json without formatting', () => {
+  const contentType = getContentType([
+    JSON.stringify({
+      foo: 'baz'
+    }),
+    JSON.stringify({
+      foo: 'bar'
+    })
+  ].join('\n'))
+
+  equal(contentType, APPLICATION_NEWLINE_JSON);
+});
+
+// not to spec, fallback to json
+test('line delimited json with formatting', () => {
+  const contentType = getContentType([
+    JSON.stringify({
+      foo: 'baz'
+    }),
+    JSON.stringify({
+      foo: 'bar'
+    }, null, 2)
+  ].join('\n'))
+
+  equal(contentType, APPLICATION_JSON);
+});
+
+test('one line of of json without formatting', () => {
+  const contentType = getContentType([
+    JSON.stringify({
+      foo: 'baz'
+    })
+  ].join('\n'))
+
+  equal(contentType, APPLICATION_NEWLINE_JSON);
+});
+
+test('json over multiple lines', () => {
+  const contentType = getContentType(JSON.stringify({
+    foo: 'bar'
+  }, null, 2));
+
+  equal(contentType, APPLICATION_JSON);
```

Comment:
x-ndjson requires each line to be a valid json object.  if there's lines that aren't parseable then we use application/json